File: l33tcode/single-row-keyboard.py
from typing import Dict


class Solution:
    def calculateTime(self, keyboard: str, word: str) -> int:
        def get_positions(keyboard: str) -> Dict[str, int]:
            positions = {}

            for pos, char in enumerate(keyboard):
                positions[char] = pos

            return positions


        moves = 0
        positions = get_positions(keyboard)
        prev_pos = 0

        for char in word:
            cur_pos = positions[char]

            moves += abs(cur_pos - prev_pos)

            prev_pos = cur_pos

        return moves

File: l33tcode/single-row-keyboard.go
package main

import "math"

func calculateTime(keyboard string, word string) int {
	cursor := 0

	keyToPos := map[rune]int{}

	for pos, char := range keyboard {
		keyToPos[char] = pos
	}

	distance := 0

	for _, char := range word {
		distance += int(math.Abs(float64(keyToPos[char]) - float64(cursor)))

		cursor = keyToPos[char]
	}

	return distance
}
